internal/backoff/backoff_test.go

```diff
@@ -472,74 +239,396 @@ func Test_backoff_Close(t *testing.T) {
 		backoffTimeLimit      time.Duration
 		errLog                bool
 	}
-	type want struct{}
+	type want struct {
+		wantRes interface{}
+		err     error
+	}
 	type test struct {
 		name       string
+		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want) error
-		beforeFunc func()
-		afterFunc  func()
+		checkFunc  func(want, interface{}, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want) error {
+	defaultCheckFunc := func(w want, gotRes interface{}, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if !reflect.DeepEqual(gotRes, w.wantRes) {
+			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", gotRes, w.wantRes)
+		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       fields: fields {
-		           wg: sync.WaitGroup{},
-		           backoffFactor: 0,
-		           initialDuration: 0,
-		           jittedInitialDuration: 0,
-		           jitterLimit: 0,
-		           durationLimit: 0,
-		           maxDuration: 0,
-		           maxRetryCount: 0,
-		           backoffTimeLimit: nil,
-		           errLog: false,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           fields: fields {
-		           wg: sync.WaitGroup{},
-		           backoffFactor: 0,
-		           initialDuration: 0,
-		           jittedInitialDuration: 0,
-		           jitterLimit: 0,
-		           durationLimit: 0,
-		           maxDuration: 0,
-		           maxRetryCount: 0,
-		           backoffTimeLimit: nil,
-		           errLog: false,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Sorry. could you please teach me why do you create cancel function?:thinking:
```suggestion
			ctxl := context.Background()
```

Reply:
Once, I thought I'll use `cancel()` in the test code.
But, actually, I did not use 😅 
I'll remove it where `cancel()` is not used.